### PR TITLE
Issue #599 - Fix payment api to allow retrieving payment transaction tags

### PR DESCRIPTION
**Pull request by javier-gomez-github**
- Added new method to get a Payment Transaction by Id

### CODE REVIEW

File: src/main/java/org/killbill/billing/payment/api/PaymentApi.java [hunk 1]
```diff
@@ -646,4 +646,13 @@ public void setDefaultPaymentMethod(Account account, UUID paymentMethodId, Itera
...
+
+    /**
+     * @param paymentTransactionId the payment transaction id
+     * @param context         the call context
+     * @return the payment transaction
+     * @throws PaymentApiException
+     */
+    public PaymentTransaction getPaymentTransactionById(final UUID paymentTransactionId, final TenantContext context)
+            throws PaymentApiException;
 }
```
Comment (sbrossie) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: I thought [we said given a transaction, we would return the payment](https://github.com/killbill/killbill/issues/599#issuecomment-249255742):

```
Payment getPaymentByTransactionId(paymentTransactionId, bool, bool, properties, context);
```

Any reason why we should return the `PaymentTransaction` instead?

Comment (javier-gomez-github) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: Yes you are right.
I thought we needed the PaymentTransaction instead, but I can change that to get the Payment.
Working on that! 📝
Thanks!

Comment (javier-gomez-github) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: @sbrossie 
Fixed and updated all Pull Requests.

Thanks!

File: src/main/java/org/killbill/billing/payment/api/PaymentApi.java [hunk 2]
```diff
@@ -646,4 +646,12 @@ public void setDefaultPaymentMethod(Account account, UUID paymentMethodId, Itera
...
             throws PaymentApiException;
-}
+
+    /**
+     * @param transactionId the payment transaction id
+     * @param context         the call context
+     * @return the payment transaction
+     * @throws PaymentApiException
+     */
+    public Payment getPaymentByTransactionId(final UUID transactionId, final boolean withPluginInfo, final boolean withAttempts, final Iterable<PluginProperty> properties, final TenantContext context) throws PaymentApiException;
```
Comment (pierre) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: Nit: move it next to the other payment-related methods above (maybe after `getPaymentByExternalKey`).

File: src/main/java/org/killbill/billing/payment/api/PaymentApi.java [hunk 3]
```diff
@@ -646,4 +646,12 @@ public void setDefaultPaymentMethod(Account account, UUID paymentMethodId, Itera
     @RequiresPermissions(PAYMENT_METHOD_CAN_UPDATE)
     public List<PaymentMethod> refreshPaymentMethods(Account account, Iterable<PluginProperty> properties, CallContext context)
             throws PaymentApiException;
-}
+
+    /**
+     * @param transactionId the payment transaction id
+     * @param context         the call context
+     * @return the payment transaction
```
Comment (pierre) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: Typo: the payment.

File: src/main/java/org/killbill/billing/payment/api/PaymentApi.java [hunk 4]
```diff
@@ -646,4 +646,12 @@ public void setDefaultPaymentMethod(Account account, UUID paymentMethodId, Itera
...
+     * @throws PaymentApiException
+     */
+    public Payment getPaymentByTransactionId(final UUID transactionId, final boolean withPluginInfo, final boolean withAttempts, final Iterable<PluginProperty> properties, final TenantContext context) throws PaymentApiException;
+    }
```
Comment (pierre) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: Nit: extra space.

File: src/main/java/org/killbill/billing/payment/api/PaymentApi.java (hunk 3 above)
Comment (javier-gomez-github) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: Fixed

File: src/main/java/org/killbill/billing/payment/api/PaymentApi.java (hunk 2 above)
Comment (javier-gomez-github) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: Fixed

File: src/main/java/org/killbill/billing/payment/api/PaymentApi.java (hunk 4 above)
Comment (javier-gomez-github) [inline src/main/java/org/killbill/billing/payment/api/PaymentApi.java]: Fixed

**Pull request merged**